### PR TITLE
Russian translation and some improvements in English text

### PR DESCRIPTION
**Pull request by eugeneloza**
Finally I've re-read through the book, seems that this variant is mature enough to be pulled. I've updated the examples with coding conventions, improved some text in English, and also made some fixes to Russian translation.
I remember that there were some extensive improvements in English version after my translation and I saw those in diffs for `modern_pascal_introduction.adoc`. But for now I cannot see them, maybe something went wrong with pull requests?

### CODE REVIEW

File: modern_pascal_introduction.adoc [hunk 1]
```diff
@@ -34,7 +34,7 @@ include::code-samples/hello_world.lpr[]
 This is a complete program that you can _compile_ and _run_.
 
 * If you use the command-line FPC, just create a new file `myprogram.lpr` and execute `fpc myprogram.lpr`.
-* If you use _Lazarus_, create a new project (menu _Project_ -> _New Project_ -> _Simple Program_). Save it as `myprogram` and paste this source code as the main file. Compile using the menu item _Run -> Compile_.
+* If you use _Lazarus_, create a new project (menu _Project_ -> _New Project_ -> _Simple Program_). Save it as `myprogram` and paste this source code as the main file. Compile using the menu item _Run -> Compile_ (F9 hotkey).
 * This is a command-line program, so in either case -- just run the compiled executable from the command-line.
```
Comment (michaliskambi) [inline modern_pascal_introduction.adoc]: Note that the hotkey is not F9 on all platforms (in particular, not on macOS). I would prefer to not specify the hotkey here -- users who use the menu item will be able to see the shortcut easily :)

**Comment by michaliskambi** [comment]
Wow, a huge change :) Thank you, I'll try to review it around the weekend.

Note 1: There were indeed changes (on my side, to English version) after your translation. Your translation is based on an older version, it seems. We have a number of conflicts because of that. I can resolve them, but if you'd like to help you can also resolve them yourself :) You can follow the procedure on https://help.github.com/articles/syncing-a-fork/ , and be aware that the "merge step" will result in conflicts. You will need to resolve and commit them.

Note 2: I wondered whether to change `WriteLn` or keep `Writeln` (lowercase L). I'm accustomed to using `Writeln` (even though `WriteLn` is more reasonable following CamelCase; I just treat `Writeln` as an exception). I tested the popularity on Lazarus sources, and your version is indeed better, the spelling `WriteLn` is more popular :) If you're curious, here are the stats:

- all lowercase `writeln` (authors probably don't care about case): 2107 occurrences
- `WriteLn` : 1480
- `Writeln` : 365
- `WriteLN` : 157
- `WRITELN` : 36
- `writeLn` : 3

Running a statistics like this may not be the best metric, as one huge file with a lot of writelns written by a single contributor could easily skew my statistics... but it seems that writelns in Lazarus are spread out across many files and many styles.

So, `WriteLn` wins :)

File: modern_pascal_introduction.adoc (hunk 1 above)
Comment (eugeneloza) [reply modern_pascal_introduction.adoc]: Fixed :)

**Comment by eugeneloza** [comment]
Yeah, and I can see now all the differences in the summary commit: https://github.com/eugeneloza/modern-pascal-introduction/commit/a86e0a9ffb130d430a727cd290f2c5126e5dadf3 --- I'll translate everything that was updated since in the nearest future.

**Comment by michaliskambi** [comment]
I am sorry that it took me so long.

Thank you, everything applied!

I did some minor changes, comments are in the commits (I edited your branch before merging, and after merging I did some additional commits, mostly minor changes to the English text in `modern_pascal_introduction.adoc`). I liked all your code style changes, and thank you for the additional text and corrections. I only changed minor things here and there.

Thank you for this gigantic work!

**Comment by eugeneloza** [comment]
Cool! And I'm still too slow to update Russian to be up-to-date with English variant :) But finally I've gotten a vacation (I hope it won't be canceled as the one in August :D), so I really hope I can finish most of the things I've started but never finished :)

**Comment by michaliskambi** [comment]
I have modified the `Makefile` commands to automatically test code in `code-samples_russian/` and to create and upload HTML and PDF versions of the Russian version.

It is available online here:

https://castle-engine.io/modern_pascal_introduction_russian.html

https://castle-engine.io/modern_pascal_introduction_russian.pdf

I'll announce these URLs next week in CGE news. These links are "stable" so feel welcome to post about them everywhere in the Russian Pascal community.

Questions:

- What are the names of the translators in English? As far as I understand, translators are "Александр Скворцов" and "Евгений Лоза". "Евгений Лоза" is "Eugene Loza", right? What is the correct spelling of "Александр Скворцов" in the English (Latin) alphabet? 

    I don't understand Russian, and I don't want to make mistakes here when crediting you both. And I don't trust Google Translate enough for this.

- Are you sure you don't want to put your names higher? I would be very happy to add your names right at the top (2nd line of the `.adoc` file), like "Michalis Kamburelis, translation: Александр Скворцов и Евгений Лоза". You did great work, and I want you to be recognized for it!

- Please pass on my thanks to the 2nd translator:) Again, much thanks, this was a big task! (And hey, it's my first book translated into a foreign language! Thanks to you.)

**Comment by eugeneloza** [comment]
Great.
However, a few notes...

> I'll announce these URLs next week in CGE news.

Maybe we can wait a bit? I'd like to finish the untranslated parts, but I'm very unsure if I can do that in a single week (of course, we can do an update after announcing too :) - so just a note).

> What are the names of the translators in English?

My name (when spelled correctly) is Yevhen Loza. I'm just used to EugeneLoza as a nickname (because I've used this one when I didn't know English yet :)).
"Александр Скворцов" should be transliterated as Aleksandr Skvortsov (according to the transliteration rules). However, in our countries our transliteration sometimes goes very wrong :D So I'd better ask him.

> Are you sure you don't want to put your names higher?

Well... I've never seen translators too high in any serious book :) Especially in fiction books, I've never actually seen the translators names at all (I'm sure they were somewhere, but I didn't see them :)). However, sometimes I've seen "translator's notes" (often speaking about how important the work is and dealing with some "translation peculiarities", e.g. some terminology).
But of course you can do as you see better :) I don't see it as "needed" - a note in the end is perfectly enough for me.

> Please pass on my thanks to the 2nd translator

Sure! I'll try to contact him right away (however, after Vkontakte has been banned in Ukraine I don't have the direct contact anymore, I'll try to get in touch through GitHub account).

> And hey, it's my first book translated into a foreign language!

You're very welcome!
If there were a lot of (at least a dozen of) Russian-speaking users of Castle Game Engine it might become plausible to translate all the documentation :) Hmm... maybe I should start translating it anyway? At least will force me actually to read it all carefully :D

**Comment by michaliskambi** [comment]
> Maybe we can wait a bit? I'd like to finish the untranslated parts, but I'm very unsure if I can do that in a single week (of course, we can do an update after announcing too :) - so just a note).

I would advice this: I can announce the Russian translation (in it's current state) next week, *and* whenever you translate the rest -- I can announce it again!

This way we do not place ourselves as a "bottleneck". I know that I often wait with announcing things until they are "perfect", and it is sometimes not optimal. In this case, I can see you already translated a *lot*, I'm sure people will like it, even if it not 100% up-to-date with the current English version.

That said, it is your choice:) If you tell me "no", then I will wait:) But I advice not to wait:)

> Well... I've never seen translators too high in any serious book :) Especially in fiction books, I've never actually seen the translators names at all (I'm sure they were somewhere, but I didn't see them :)). 

In Polish books, the translator name is often quite prominent. Not as prominent as the title or the original author, but still it's usually clearly visible, often at the bottom of the "title page".

In any case, who cares about the conventions? :) You did a large work, and you did it for free in your spare time, and everyone can read it for free too. If my name is up there, yours should be too :)

**Comment by eugeneloza** [comment]
Ok, let's do it! :D

**Pull request merged**